### PR TITLE
feat: markdown editor

### PR DESCRIPTION
**Pull request by cabreraalex**

### CODE REVIEW

File: src/lib/components/prompt/RewriteSuggestion.svelte [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	}
+</script>
+
+<div
+	class="flex w-full flex-col justify-between rounded-br rounded-tr border border-l-4 border-l-blue-600 px-3 py-2 text-left"
+	onmouseup={(e) => e.stopPropagation()}
+>
+	<div class="flex items-center gap-2">
+		<FlaskConical class="h-5 w-5 text-blue-600" />
+		<h4>Rewrite the selected text</h4>
```
Comment (Sparkier) [inline src/lib/components/prompt/RewriteSuggestion.svelte:41]: This has been updated.

File: src/lib/components/prompt/RewriteSuggestion.svelte [hunk 2]
```diff
@@ -0,0 +1,61 @@
+<script lang="ts">
```
[the remark this answers is not in the document]
Comment (Sparkier) [reply src/lib/components/prompt/RewriteSuggestion.svelte:1]: maybe separate PR?

**Review by Sparkier — CHANGES_REQUESTED** [review]

File: src/lib/server/editors/MarkdownEditor.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
+import type { LLM } from '$lib/server/llms/llm';
+import { fetchPrompt } from '$lib/server/prompts.js';
+import type { Tables } from '$lib/supabase';
+import { EditorType, RequiredInputType } from '$lib/types';
+import { PromptEditor } from './editor';
+
+const filterPrompt = await fetchPrompt('4GznJqeM', '789');
+const applyPrompt = await fetchPrompt('iXgXi2Wk', '788');
```
[the remark this answers is not in the document]
Comment (cabreraalex) [reply src/lib/server/editors/MarkdownEditor.ts:8]: shared!

File: src/lib/server/editors/MarkdownEditor.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+			'Structure your prompt using markdown.',
+			EditorType.ENHANCEMENT,
+			RequiredInputType.TEXT
+		);
+	}
+
+	async canBeApplied(prompt: Tables<'prompts'>, llm: LLM) {
+		if (prompt.responseFormat === 'json') {
+			return null;
+		}
```
[the remark this answers is not in the document]
Comment (cabreraalex) [reply src/lib/server/editors/MarkdownEditor.ts:24]: good catch, from copying over.

File: src/lib/components/prompt/RewriteSuggestion.svelte (hunk 2 above)
[the remark this answers is not in the document]
Comment (cabreraalex) [reply src/lib/components/prompt/RewriteSuggestion.svelte]: yeah sorry included by accident

**Review by Sparkier — APPROVED** [review]
lgtm generally but this is not in the editors list yet, right? Also not in hypertune.

**Pull request merged**